test/README.md

```diff
@@ -0,0 +1,179 @@
+release...
+
+```
+export KNATIVE_EVENTING_RELEASE_OVERRIDE="v0.17.3"
+```
+
+### GO Test Cmd
+
+This is the fastest way to iterate when developing integration tests.  It does require, though,
+that you provide the K8S Cluster pre-configured with everything needed to run the tests.  This
+includes Knative-Eventing, a Kafka Cluster, and the KafkaChannel CR implementation.  Your
+current `kubeconfig` should refer to the cluster.  Further, you are responsible for providing
+the Knative Eventing [Test Images](#test-images) which are normally built for you when running
+the [e2e-tests.sh](./e2e-tests.sh) script.
+
+```
+# Run All The Conformance Tests Against The KafkaChannel
+go test -v -tags=e2e -count=1 ./test/conformance/... -channels=messaging.knative.dev/v1beta1:KafkaChannel
+
+# Run All The E2E Tests Against The KafkaChannel
+go test -v -tags=e2e -count=1 ./test/e2e/... -channels=messaging.knative.dev/v1beta1:KafkaChannel
+
+# Run Only The 'TestChannelChain' E2E Test Against The KafkaChannel
+go test -v -tags=e2e -count=1 ./test/e2e/... -channels=messaging.knative.dev/v1beta1:KafkaChannel -run TestChannelChain
+```
+
+> Note - The `-tags=e2e` argument is required to run the integration tests with `go test`!
+
+## Performance Tests
+
+> TODO - Implement Performance Tests ; )
+
+## Test Images
+
+The integration tests primarily delegate to the Knative Eventing generic test implementations.
+These implementations use various [test_images](../vendor/knative.dev/eventing/test/test_images)
+in order to verify the desired behavior.  Therefore, these `test_images` need to have been built
+and be available for downloading in the cluster.  The [e2e-tests.sh](./e2e-tests.sh) script will
+build and publish these for you, but you are responsible for providing them when running the
+tests directly via `go test` commands.
+
+To manually build and publish the test_images to `$KO_DOCKER_REPO` from the project root...
+
+```
+export KO_DOCKER_REPO=<my-docker-repo>
+export VENDOR_EVENTING_TEST_IMAGES="vendor/knative.dev/eventing/test/test_images/"
+sed -i 's@knative.dev/eventing/test/test_images@knative.dev/eventing-kafka/vendor/knative.dev/eventing/test/test_images@g' "${VENDOR_EVENTING_TEST_IMAGES}"*/*.yaml
+./test/upload-test-images.sh vendor/knative.dev/eventing/test/test_images/ e2e
+sed -i 's@knative.dev/eventing-kafka/vendor/knative.dev/eventing/test/test_images@knative.dev/eventing/test/test_images@g' "${VENDOR_EVENTING_TEST_IMAGES}"*/*.yaml
+```
+
+>Note - Set `$KO_DOCKER_REPO` to whatever Docker Repository you intend to use.  You cannot use
+>`ko.local` for the Knative Eventing `test_images` because they will be deployed with an
+>`ImagePullPolicy` of `Always` which will bypass the local docker repo.
+>See [Private Docker Repositories](#private-docker-repositories) for further information.
+>Note - See [MacOS Utils](#macos-utils) when running on Mac and replace `sed` in above commands
+>with `gsed`.
```

Comment:
<!--mdfmt-bot-->
Format markdown:
```suggestion
> Note - Set `$KO_DOCKER_REPO` to whatever Docker Repository you intend to use.
> You cannot use `ko.local` for the Knative Eventing `test_images` because they
> will be deployed with an `ImagePullPolicy` of `Always` which will bypass the
> local docker repo. See
> [Private Docker Repositories](#private-docker-repositories) for further
> information. Note - See [MacOS Utils](#macos-utils) when running on Mac and
> replace `sed` in above commands with `gsed`.
```